src/snowflake/snowpark/mock/_functions.py

```diff
@@ -469,15 +469,21 @@ def mock_to_time(
     try_cast: bool = False,
 ):
     """
-    [ ] For string_expr, the result of converting the string to a time.
+    [x] For string_expr, the result of converting the string to a time.
 
-    [ ] For timestamp_expr, the time portion of the input value.
+    [x] For timestamp_expr, the time portion of the input value.
 
-    [ ] For 'integer' (a string containing an integer), the integer is treated as a number of seconds, milliseconds, microseconds, or nanoseconds after the start of the Unix epoch. See the Usage Notes below.
+    [x] For 'integer' (a string containing an integer), the integer is treated as a number of seconds, milliseconds, microseconds, or nanoseconds after the start of the Unix epoch. See the Usage Notes below.
 
-    [ ] For this timestamp, the function gets the number of seconds after the start of the Unix epoch. The function performs a modulo operation to get the remainder from dividing this number by the number of seconds in a day (86400): number_of_seconds % 86400
+        [x] For this timestamp, the function gets the number of seconds after the start of the Unix epoch. The function performs a modulo operation to get the remainder from dividing this number by the number of seconds in a day (86400): number_of_seconds % 86400
 
     """
+
+    def convert_int_string_to_time(d: str):
+        return datetime.datetime.utcfromtimestamp(
+            process_numeric_time(d) % 86400
```

Comment:
Seems like `process_numeric_time` converts integer values that could be milliseconds or nanoseconds to seconds? Does it handle microseconds? 

NIT: Can we rename the function to something like `convert_integer_value_to_seconds`?

Reply:
updated to handle all unit as per the spec, also added tests